packages/aws-cdk-lib/aws-cognito/lib/user-pool-domain.ts

```diff
@@ -114,18 +115,29 @@ export class UserPoolDomain extends Resource implements IUserPoolDomain {
     this.isCognitoDomain = !!props.cognitoDomain;
 
     const domainName = props.cognitoDomain?.domainPrefix || props.customDomain?.domainName!;
-    const resource = new CfnUserPoolDomain(this, 'Resource', {
+    this.resource = new CfnUserPoolDomain(this, 'Resource', {
       userPoolId: props.userPool.userPoolId,
       domain: domainName,
       customDomainConfig: props.customDomain ? { certificateArn: props.customDomain.certificate.certificateArn } : undefined,
     });
 
-    this.domainName = resource.ref;
+    this.domainName = this.resource.ref;
   }
 
   /**
    * The domain name of the CloudFront distribution associated with the user pool domain.
    */
+  public get cloudFrontEndpoint(): string {
```

Comment:
This method name is based on the description in CFn doc.

> CloudFrontDistribution
> The Amazon CloudFront endpoint that you use as the target of the alias that you set up with your Domain Name Service (DNS) provider.

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-cognito-userpooldomain.html#aws-resource-cognito-userpooldomain-return-values